docs/adr/0003-delivery-channels-templating.md

```diff
@@ -0,0 +1,88 @@
+
+Creation of templates can be totally flexible and dependent on the organisation which will integrate with `Marain.UserNotifications` thus every tenant will have their own set of templates for SMS, Email etc. if that organisation wishes to use that communication channel and would not need to setup templates for the unused communication channels.
+
+So, if a notification is created but its corresponding configuration is not setup for that tenant and its communication type, then there will be no notification triggered and an error would be logged in the existing notification data table (azure storage table) which would alert that there is some missing configuration for the newly added notification type for the choosen delivery channel.
+
+### Storage of templates
+
+Templates need to be stored in such a way where it can be accessed and modified easily on the fly. Thus to have this flexibilty, templates will be stored in azure table storage. The template for individual properties (`Body`, `Subject`) would have the following column headers:
+
+- Notification Type
+- Communication Type
+- Property Name (This will be the property name of the communication object)
+- Template
+
+Eg. The below example shows how the table storage will host the relevant templates mentioned above having a template for every single property name per communication type and notification type.
+
+```
+Notification Type       Communication Type         Property Name        Template
+
+New Lead                SMS                        Body                 Hi {{endUserName}},
+                                                                        You have been assigned a new lead from {{fromUserName}}.
+
+New Lead                Email                      Body                 Hi {{endUserName}},
+                                                                        You have been assigned a new lead from {{fromUserName}} having name {{leadName}}.
+
+New Lead                Email                      Subject              New Lead Assigned
+
+```
+
+### Applying a certain template to a property for a communication object
+
+The Management API will be responsible to convert the create notification request into a particular communication object (SMSObject) based on the configured user preferences for the end user. Every communication object has certain required properties without which a notification cannot be sent as explained in the section above. If we arrive in such a scenario where the essential properties are missing then the `ChannelDeliveryStatusesJson` column within the existing notification data table (azure storage table) would be updated to specify which communication type was not successfully delivered with a message including a reason.
```

Comment:
This will no longer be "based on the configured user preferences for the end user" - it'll be based on the data supplied with the create notification request.